pkg/asset/agent/image/ignition.go

```diff
@@ -360,6 +373,53 @@ func addExtraManifests(config *igntypes.Config, extraManifests *manifests.ExtraM
 	}
 }
 
+func getOSImagesInfo() (*models.OsImage, error) {
+	ctx, cancel := context.WithTimeout(context.TODO(), 30*time.Second)
+	defer cancel()
+
+	st, err := rhcos.FetchCoreOSBuild(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// Architecture currently hard-coded to x86_64
+	cpurArch := archName
+	osImage := &models.OsImage{
+		CPUArchitecture: &cpurArch,
+	}
+
+	ver, err := version.Version()
+	if err != nil {
+		return nil, err
+	}
+	osImage.OpenshiftVersion = &ver
+
+	streamArch, err := st.GetArchitecture(archName)
+	if err != nil {
+		return nil, err
+	}
+
+	artifacts, ok := streamArch.Artifacts["metal"]
+	if !ok {
+		return nil, fmt.Errorf("unable to retrieval coreos metal info for architecture %s", archName)
+	}
+	osImage.Version = &artifacts.Release
+
+	isoFormat, ok := artifacts.Formats["iso"]
+	if !ok {
+		return nil, fmt.Errorf("unable to retrieval coreos ISO info for architecture %s", archName)
+	}
+	osImage.URL = &isoFormat.Disk.Location
```

Comment:
We already have code to retrieve this (in the download fallback if we can't extract the image from the payload), I wonder if we could reuse it here.